src/gromacs/nblib/topology.cpp

```diff
@@ -182,8 +298,41 @@ gmx::ListOfLists<int> TopologyBuilder::createExclusionsListOfLists() const
     return exclusionsListOfListsGlobal;
 }
 
+Topology::InteractionData TopologyBuilder::createInteractionData(const detail::ParticleSequencer& particleSequencer)
+{
+    auto create = [](auto& output, const auto& molecules, const auto& sequencer) {
+        using BondType = typename std::decay_t<decltype(output)>::type;
+        auto uniqueData = detail::eliminateDuplicateBonds(detail::aggregateBonds<BondType>(molecules));
+        auto& uniqueIndices       = std::get<0>(uniqueData);
+        auto& uniqueBondInstances = std::get<1>(uniqueData);
+
+        // add data about BondType instances
+        output.bondInstances = uniqueBondInstances;
+
+        // add data about interaction pair indices
+        auto& indexTriples = output.indices;
```

Comment:
Where does this come from? How is output getting set?

Reply:
Yeah, might as well call it "interactionData". If you want to disambiguate the lambda function parameter from the local variable in the surrounding scope, you could call the local variable "interactionData_".

Reply:
has been addressed